codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/protocols/ServerHttpBoundProtocolGenerator.kt

```diff
@@ -699,6 +724,17 @@ private class ServerHttpBoundProtocolTraitImplGenerator(
         val err = if (StructureGenerator.fallibleBuilder(inputShape, symbolProvider)) {
             "?"
         } else ""
+        if (noInputs) {
+            val emptyBody = if (parser != null) { "body.is_empty()" } else { "true" }
+            rustTemplate(
+                """
+                #{SmithyHttpServer}::protocols::content_type_header_empty_body_no_modeled_input(
```

Comment:
I think we should split this code. There should be two functions, each handling the case of whether the body is empty or not.

Reply:
But when the body is not empty, it's just `Ok(())`